Feature null reaction dir

EnnaDelfen opened the pull request:
This branch reinstate the "Null" reaction folder so that when "no chemistry" is needed (in the sense that no reactions occur, not necessarily mean we have no species) then react() simply does nothing.
Makes it possible to compile non reacting test cases without sundials.
Does not break anything or changes any structure in place currently in any of the Pele codes

Commit:
Specifying a ChemistryDir shouldn't be optional in the current state of things

Commit:
Add a Null folder under Reactions for when we want to bypass Sundials/dvode and all other chem integration

Commit:
Fix a few indentation issues

Commit:
Few includes in Null/reactor* missing
Modify GNUmakefiles of test cases to Reaction=Null when needed
A few more indentation issues in main.cpp of some test cases (all actually)

drummerdoc commented on Testing/Exec/Make.PelePhysics:
```diff
@@ -71,7 +71,7 @@ ifeq ($(USE_F90_PP),TRUE)
     VPATH_LOCATIONS   += $(REACTIONS_HOME_F90) $(REACTIONS_PATH_F90)
 endif
-ifdef Chemistry_Model
+#ifdef Chemistry_Model
```
Seems weird to have to use defines in GNUmakefiles

EnnaDelfen replied:
Yes, I commented them out in case but I'll remove them completely.

EnnaDelfen commented:
Waiting on Lucas to test it on a PLM case

Commit:
Remove the ifdef ChemistryDir in Make.PelePhysics

Commit:
BugFix in include locations

Commit:
A few more "bugfix". We need to integrate the external source term, and
recompute the T.
Gives about same results than hacknochem but need to check convergence

esclapez reviewed (approved):
Looks fine to me. Gives the same results as going through hack_nochem in PeleLM (but HR is not hard zero because of machine precision operations).
